recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch

```diff
@@ -0,0 +1,485 @@
++	sdctl |= SC_RSTB_PLL_MASK;
++	writel(sdctl, serdes_ctl_reg);
++
++	udelay(1000);
++
++	return;
++}
++#endif /* IS_ENABLED(CONFIG_ARCH_XGS_IPROC) */
++
+ static int platform_phy_connect(struct bgmac *bgmac)
+ {
+ 	struct phy_device *phy_dev;
+ 
+-	if (bgmac->plat.nicpm_base)
++	if (bgmac->plat.nicpm_base) {
+ 		phy_dev = of_phy_get_and_connect(bgmac->net_dev,
+ 						 bgmac->dev->of_node,
+ 						 bgmac_nicpm_speed_set);
+-	else
++	}
++	else {
++		struct device_node *np = bgmac->dev->of_node;
++		u32 lane;
++
++		/* For WH2 SGMII case, treat SERDES as PHY */
++		if (of_device_is_compatible(np, "brcm,xgs-wh2-amac") &&
++			is_wh2_amac_sgmii()) {
++			struct device_node *phy_np;
++			phy_interface_t iface = PHY_INTERFACE_MODE_SGMII;
++
++			phy_np = of_parse_phandle(np, "serdes-phy-handle", 0);
++			if (!phy_np)
++				return -ENODEV;
++
++			phy_dev = of_phy_find_device(phy_np);
++			if (!phy_dev)
++				return -ENODEV;
++
++			/* Get lane from DT, otherwise set to default 3 */
++			if (of_property_read_u32(phy_np, "lane-num", &lane))
++				lane = 3;
++			xgs_serdes_set_lane(phy_dev, lane);
++
++			amac_serdes_init(bgmac, phy_dev);
++
++			phy_connect_direct(bgmac->net_dev, phy_dev,
++						bgmac_adjust_link, iface);
++
++			put_device(&phy_dev->mdio.dev);
++			of_node_put(phy_np);
++		}
++		else {
+ 		phy_dev = of_phy_get_and_connect(bgmac->net_dev,
+ 						 bgmac->dev->of_node,
+ 						 bgmac_adjust_link);
++		}
++	}
++
+ 	if (!phy_dev) {
+ 		dev_err(bgmac->dev, "PHY connection failed\n");
+ 		return -ENODEV;
+@@ -283,6 +401,8 @@ static const struct dev_pm_ops bgmac_pm_ops = {
+ static const struct of_device_id bgmac_of_enet_match[] = {
+ 	{.compatible = "brcm,amac",},
+ 	{.compatible = "brcm,nsp-amac",},
++	{.compatible = "brcm,xgs-iproc-amac",},
++	{.compatible = "brcm,xgs-wh2-amac",},
+ 	{.compatible = "brcm,ns2-amac",},
+ 	{},
+ };
+diff --git a/drivers/net/ethernet/broadcom/bgmac.c b/drivers/net/ethernet/broadcom/bgmac.c
+index 5fb3af5670ec..180ec19ce8fb 100644
+--- a/drivers/net/ethernet/broadcom/bgmac.c
++++ b/drivers/net/ethernet/broadcom/bgmac.c
+@@ -18,6 +18,12 @@
+ #include <net/dsa.h>
+ #include "bgmac.h"
+ 
++#if IS_ENABLED(CONFIG_ARCH_XGS_IPROC)
++#include <linux/of_mdio.h>
++#include <linux/phy/xgs_iproc_serdes.h>
++extern void amac_serdes_init(struct bgmac *, struct phy_device *);
++#endif
++
+ static bool bgmac_wait_value(struct bgmac *bgmac, u16 reg, u32 mask,
+ 			     u32 value, int timeout)
+ {
+@@ -1118,6 +1124,7 @@ static void bgmac_chip_init(struct bgmac *bgmac)
+ 
+ 	bgmac_umac_write(bgmac, UMAC_MAX_FRAME_LEN, 32 + ETHER_MAX_LEN);
+ 
++	if (!IS_ENABLED(CONFIG_ARCH_XGS_IPROC))
+ 	bgmac_chip_intrs_on(bgmac);
+ 
+ 	bgmac_enable(bgmac);
+@@ -1195,10 +1202,25 @@ static int bgmac_open(struct net_device *net_dev)
+ 	}
+ 	napi_enable(&bgmac->napi);
+ 
++	if (IS_ENABLED(CONFIG_ARCH_XGS_IPROC)) {
++		/* Reset emulated MIB statistics to zero */
++		memset(&bgmac->estats, 0, sizeof(struct bgmac_ethtool_stats));
++	}
```

Comment:
I see no mention in the commit message of this change compared to the Linux 5.15 version.

Reply:
As indicated by the single pluses, it's a change in the upstream driver, not in the modifications of the patch.

Reply:
That change was apparently accidentally dropped in https://github.com/bisdn/meta-open-network-linux/commit/351e020ee9c77df4c4b2abbee80a3343ee1fbf59.

Reply:
So shouldn't adding it back be a separate commit which fixes the accident for 5.15 as well?

Reply:
I didn't intentionally add it back, I will need to figure out where it got lost for 5.15.

Reply:
Might have been a quick (build) fix I forgot to replace with the actual commit.